### DISCUSSION TITLE
Second monitor is blank when resuming from suspend

### DISCUSSION THREAD

**Discussion by smileservices** [i3/i3#6162] [Q&A]
**Setup**
- lenovo x1 carbon gen 10
- 2nd monitor linked through hdmi cable
- ubuntu 22.04, i3wm, i3lock, i3lock-fancy

**What's wrong**
 After resuming from suspend, the 2nd monitor is blank, not receiving any signal from the HDMI, although `xrandr --listmonitors` shows both the main and the hdmi connected one. Running `xrandr --auto` does nothing as well. The only workaround is to open the 2nd monitor, then resume suspend.

When using the ubuntu desktop enviroment the suspend/resume works without problem, so i'm suspecting it's from either i3wm or i3lock. Any ideas to solve this, please?

**Comment by orestisfl** [comment]
> When using the ubuntu desktop enviroment the suspend/resume works without problem, so i'm suspecting it's from either i3wm or i3lock. Any ideas to solve this, please?

i3/i3lock don't interact with your monitors in this kind of way so the most likely explanation is ubuntu desktop doing something that xrandr doesn't do by itself.

Try saving your layout with arandr and apply it after sleep. Do the monitors wake up?

**Comment by KebabLord** [comment]
You can also try disabling your composer (picom or compton) before suspend to see if they're the reason
